test/parallel/test-assert-strict-typecheck.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+// The following deepEqual tests might seem very weird.
+// They just describe what it is now.
+// That is why we discourage using deepEqual in our own tests.
+
+const date = new Date('2016');
+
+// See #10258
+function FakeDate() {}
+FakeDate.prototype = Date.prototype;
+const fake = new FakeDate();
+
+/* eslint-disable no-restricted-properties */
+// We are testing deepEqual!
```

Comment:
@joyeecheung Nope, and to be clear, I wasn’t requesting you to change anything here, just making sure that I understood why the line is there in the first place